pkg/cluster/inventory.go

```diff
@@ -48,12 +48,12 @@ func NewInventory(conn db.Connection, debug bool, collector metricsCollector) (I
 	return &DefaultInventory{repo, collector}, nil
 }
 
-func (i *DefaultInventory) CountRetries(runtimeID string, configVersion int64) (int, error) {
+func (i *DefaultInventory) CountRetries(runtimeID string, configVersion int64, maxRetries int) (int, error) {
 	q, err := db.NewQuery(i.Conn, &model.ClusterStatusEntity{}, i.Logger)
 	if err != nil {
 		return 0, err
 	}
-	clusterStatuses, err := q.Select().Where(map[string]interface{}{"RuntimeID": runtimeID, "ConfigVersion": configVersion}).OrderBy(map[string]string{"ID": "desc"}).Limit(50).GetMany()
+	clusterStatuses, err := q.Select().Where(map[string]interface{}{"RuntimeID": runtimeID, "ConfigVersion": configVersion}).OrderBy(map[string]string{"ID": "desc"}).Limit(maxRetries * 5).GetMany()
```

Comment:
To be honest, I still not understand why `*5`, is it just to increase the safeness to query enough records? 
And I have some opinion to set the Limit, I think the purpose here is performance concern, but there is not so many records in our case, and use `limit` is not a magic way, it needs to combine with proper index and ordering, otherwise the performance will even be decreased, like mentioned here:
https://stackoverflow.com/questions/6037843/extremely-slow-postgresql-query-with-order-and-limit-clauses

Regarding performance improvement, a general approach is to prevent doing overengineering, we anyway can resolve it when it actually hit performance alert to us.

I suggest deleting this `Limit` clause.

Reply:
This is a very good point! I also think that just arbitrarily multiplying by a magic number is no solution. Removing the limit isn't a solution either, we need to build a system that is scalable and we could potentially be managing 1000s of clusters when this goes live.

Instead we should see when we really need a limit, to which extent and why. My proposal would be to add some intelligence to the query builder to automatically set this depending on the query (has where? has order? is selecting by PK? Joins?)